### DISCUSSION TITLE
errors encountered in parcel 2 beta.1

### DISCUSSION THREAD

**Discussion by bugzpodder** [parcel-bundler/parcel#4762] [General]
Ran into a variety of issues when testing out parcel2-beta.1

Installed some parcel packages, it defaulted to alpha@3.2 rather than @next/beta.1, causing some weird errors when trying start parcel.  Adding @next tag and reinstalling solved it.
```@parcel/config-default @parcel/runtime-react-refresh @parcel/transformer-react-refresh-babel @parcel/transformer-react-refresh-wrap @parcel/validator-typescript```

I disabled build scripts (b/c of yarn 2) and got a non-descriptive error for @parcel/watch
```
.yarn/cache/bindings-npm-1.5.0-77ce1d213c-3.zip/node_modules/bindings/bindings.js:135
  throw err;
  ^

Error: Could not locate the bindings file. Tried:
.yarn/unplugged/@parcel-watcher-npm-2.0.0-alpha.5-fd07f40ca1/node_modules/@parcel/watcher/build/watcher.node
 ```
enabled build scripts and it worked fine.

I added a .parcelrc (to get  @parcel/transformer-raw)  

Got a good build but it says Page not found:
```
 yarn parcel ./src/index.html
ℹ️ Server running at http://localhost:1234
✨ Built in 779ms

404
Page not found.
```

and I removed .parcelrc removed .parcelrc, but parcel just read from cache and didn't rebuild from scratch based on new config file.  had to rm -rf .parcel_cache (would appreciate some CLI command)

I had a babel.config.js and got an error for react-refresh:
```
React Refresh Babel transform should only be enabled in development environment. Instead, the environment is: "production". If you want to override this check, pass {skipEnvCheck: true} as plugin options.
```
removing babel.config.js worked.

got this error, couldn't find a workaround
```
/.yarn/cache/assert-npm-2.0.0-ef73bc19f5-3.zip/node_modules/assert/build/assert.js:41:30
  40 |
> 41 | var _require$types = require('util/').types,
>    |                              ^^^^^^^
  42 |     isPromise = _require$types.isPromise,
  43 |     isRegExp = _require$types.isRegExp;
@parcel/resolver-default: The "path" argument must be of type string. Received type object
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:1147:7)
```
```
yarn why assert
├─ @parcel/node-libs-browser@npm:2.0.0-beta.1
│  └─ assert@npm:2.0.0 (via npm:^2.0.0)
│
├─ adjust-sourcemap-loader@npm:2.0.0
│  └─ assert@npm:1.4.1 (via npm:1.4.1)
│
└─ node-libs-browser@npm:2.2.1
   └─ assert@npm:1.5.0 (via npm:^1.1.1)
```

yarn: 2.0.0-rc.33
node: v10.20.1
macos 10.15.5

**Comment by DeMoorJasper** [comment]
The issue with versioning is unfortunately just how npm and yarn work...

The issue with the watcher has a solution in an open PR: https://github.com/parcel-bundler/watcher/pull/34

It should probably detect `.parcelrc` changes, that seems like a new bug

@mischnic any idea why react-fast-refresh fails?

**Reply by bugzpodder** [reply]
There is a line here that says babel.config.js should be avoided
https://v2.parceljs.org/languages/babel/

**Reply by DeMoorJasper** [reply]
Yes, it's terrible for caching that's why it should be avoided, still shouldn't throw that error though.

**Comment by mischnic** [comment]
> got this error

There is no error message in that snippet.

> I had a babel.config.js and got an error for react-refresh:

That is technically a warning but it should still not happen.

<br>

Using a babel.config.json in a monorepo works fine for me. Can you share a reproduction?

**Reply by bugzpodder** [reply]
I sometimes randomly get the following error (parcel worked in one commandline tab but failed to build in anther tab with multiple attempts), so weird.  I found that removing .babel_cache would unblock me.

```
ℹ️ Server running at http://localhost:1234
🚨 Build failed.
@parcel/core: Failed to resolve '../node_modules/@parcel/transformer-react-refresh-wrap/lib/helpers/helpers.js' from './src/index.tsx'
@parcel/resolver-default: Cannot load file '../node_modules/@parcel/transformer-react-refresh-wrap/lib/helpers/helpers.js' in './src'.
```

For babel.config.js, this is what i have, i just repo'ed it again with the same error again in two different environments (create a blank ts project in CRA and fix svg import):
```
module.exports = {
  presets: ["@babel/preset-react", "@babel/preset-typescript"],
  plugins: [
    "babel-plugin-macros",
    "babel-plugin-transform-inline-environment-variables",
  ],
  env: {
    development: {
      plugins: ["react-refresh/babel"],
    },
  },
};
```



>    got this error
> There is no error message in that snippet.

hmm it's the pastes right under it:
```
/.yarn/cache/assert-npm-2.0.0-ef73bc19f5-3.zip/node_modules/assert/build/assert.js:41:30
  40 |
> 41 | var _require$types = require('util/').types,
>    |                              ^^^^^^^
  42 |     isPromise = _require$types.isPromise,
  43 |     isRegExp = _require$types.isRegExp;
@parcel/resolver-default: The "path" argument must be of type string. Received type object
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:1147:7)

yarn why assert
├─ @parcel/node-libs-browser@npm:2.0.0-beta.1
│  └─ assert@npm:2.0.0 (via npm:^2.0.0)
│
├─ adjust-sourcemap-loader@npm:2.0.0
│  └─ assert@npm:1.4.1 (via npm:1.4.1)
│
└─ node-libs-browser@npm:2.2.1
   └─ assert@npm:1.5.0 (via npm:^1.1.1)
```

**Reply by bugzpodder** [reply]
Also, here is the browserslist
```
 "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
```

for react-refresh the error was
```
Error: [BABEL] /src/index.tsx: React Refresh Babel transform should only be enabled in development environment. Instead, the environment is: "production". If you want to override this check, pass {skipEnvCheck: true} as plugin options. (While processing: ".yarn/cache/react-refresh-npm-0.8.3-9b3fb3900b-38f06df8e0.zip/node_modules/react-refresh/babel.js")
```

**Reply by mischnic** [reply]
(Everything in that babel config is already applied by default by Parcel)

I've created a CRA app, removed package.json & installed `parcel@next react react-dom`, adjusted public/index.html to include a script tag to src/index.tsx and ran `parcel public/index.html`.  But didn't get that error

**Reply by mischnic** [reply]
EDIT: having this
```
import assert from "assert";
console.log(assert)
```
does throw that error

**Reply by bugzpodder** [reply]
> (Everything in that babel config is already applied by default by Parcel)

I agree, i was migrating an existing app where this file was pre-existing and it happened.

> I've created a CRA app, removed package.json & installed parcel@next react react-dom, adjusted public/index.html to include a script tag to src/index.tsx and ran parcel public/index.html. But didn't get that error

I reverted back to yarn 1.22 and the error still happened.  I can give it another try later.
```
my-app % yarn parcel src/index.html
yarn run v1.22.4
/my-app/node_modules/.bin/parcel src/index.html
ℹ️ Server running at http://localhost:1234
🚨 Build failed.
@parcel/transformer-babel: [BABEL] /src/index.tsx: React Refresh Babel transform should only be enabled in development environment. Instead, the environment is: "production". If you want to override this check, pass {skipEnvCheck: true} as plugin options. (While processing: "/my-app/node_modules/react-refresh/babel.js")
Error: [BABEL] /my-app/src/index.tsx: React Refresh Babel transform should only be enabled in development environment. Instead, the environment is: "production". If you want to override this check, pass {skipEnvCheck: true} as plugin options. (While processing: "/my-app/node_modules/react-refresh/babel.js")
```

**Reply by bugzpodder** [reply]
> EDIT: having this

> import assert from "assert";
> console.log(assert)

> does throw that error

Yes, I determined it only happens with yarn 2 (berry)

**Reply by bugzpodder** [reply]
minimal repo for babel.config.js:
> yarn create react-app foo
> cd foo
> yarn add parcel@next @babel/preset-react react-refresh

babel.config.js
```
module.exports = {
  presets: ["@babel/preset-react"],
  env: {
    development: {
      plugins: ["react-refresh/babel"],
    },
  },
};
```

 src/index.html
```
<!DOCTYPE html>
<html>
  <body>
    <div id="root"></div>
    <script src="./index.js"></script>
  </body>
</html>
```

Fix the svg import in App.js
>  yarn parcel src/index.html
```
ℹ️ Server running at http://localhost:1234
🚨 Build failed.
@parcel/transformer-babel: [BABEL] foo/src/index.js: React Refresh Babel transform should only be enabled in development environment.
```
